packages/tooling/fast-tooling/src/web-components/html-render/html-render.ts

```diff
@@ -124,6 +127,48 @@ export class HTMLRender extends FoundationElement {
                     );
                 }
             }
+            if (
+                e.data.type === MessageSystemType.custom &&
+                e.data.options &&
+                e.data.options.originatorId !== this.messageOriginatorId
+            ) {
+                const action: string[] = (e.data.options.action as string).split("::");
+                if (action[0] === "displayMode") {
+                    this.interactiveMode = action[1] !== "preview";
+                    if (this.interactiveMode) {
+                        this.updateLayers(
+                            this.layerActivityId,
+                            ActivityType.releaseFocus,
+                            null,
+                            null,
+                            null
+                        );
+                        // give everything time to update positions
```

Comment:
In the future we might want to consider listening for DOM updates and having a setTimeout that triggers a certain amount of time after the last update has been fired.

Reply:
Agreed.